renderer.php

```diff
@@ -84,6 +84,10 @@ public function display_file_validation($filename, local_moodlecheck_file $file,
             $output .= $filename. "\n";
         }
         foreach ($errors as $error) {
+            // Add the severity always to both text and html formats.
+            if ($format == 'html' || $format == 'text') {
+                $error['message'] .= ' (' . $error['severity'] . ')';
+            }
             if (($format == 'html' || $format == 'text') && isset($error['line']) && strlen($error['line'])) {
                 $error['message'] = get_string('linenum', 'local_moodlecheck', $error['line']). $error['message'];
             }
```

Comment:
I was very confused about why `$error['message']` being completely overwritten. Also, small cleanups because `strlen()` is a confusing way to check that the line number is not an empty string.
```suggestion
            // Add the severity always to both text and html formats.
            if ($format == 'html' || $format == 'text') {
                $error['message'] .= ' (' . $error['severity'] . ')';
            }

            // Prepend the line number, if available.
            if (($format == 'html' || $format == 'text') && !empty($error['line'])) {
                $error['message'] = get_string('linenum', 'local_moodlecheck', $error['line']) . $error['message'];
            }
```

Reply:
Yeah, in fact initially I reworked all those if blocks way more. But at the end decided to go with the minimal patch (avoiding noise).

In any case... while I agree about `isset() + strlen()` being "strange"... I'm worried about changing it to `empty()`. I remember some checks - must confess that I'm not sure if that was here, in local_moodlecheck or in other checks - where we return zero (0) as line for problems detected and not related to an exact line. 

And zero is empty... so... just being lazy... I'd propose not to change that. Other tools that post-process that information (CiBoT, local_ci, moodle-plugin-ci...) may have dependencies built on those expected zeros... (sort of don't break what is not broken).

Does that sound acceptable?

Reply:
Ok, have added your latest version as extra commit. I think that one is safe (strict) enough. Thanks!